### PR TITLE
fix: add retry and wait for kubernetes creation/deletion

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Add retry and wait logic for all kubernetes operations.

When submitting statefulset deletion call, the metadata in kubernetes is not updated immediately especially if the statefulset has many replicas.  We need to delete and wait to make sure the statefulset is delete before continuing. 

If we don't do this then stop and then start operations will no produce the correct behavior and result in statefulsets being deleted and not started back up because kuberntes has not updated the metadata for deletion prior to the creation call for the statefulset is issued

I also retry logic to cleanup subscriptions created by functions after the function is deleted

### CODE REVIEW

**Comment by jerrypeng** [comment]
@cckellogg please take a look as well

**Comment by jerrypeng** [comment]
rerun java8 tests

File: pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/KubernetesRuntime.java [hunk 1]
```diff
@@ -397,75 +419,298 @@ private void submitStatefulSet() throws Exception {
...
+                .sleepBetweenInvocationsMs(SLEEP_BETWEEN_RETRIES_MS * 2)
+                .supplier(() -> {
+                    String labels = String.format("tenant=%s,namespace=%s,name=%s",
+                            instanceConfig.getFunctionDetails().getTenant(),
+                            instanceConfig.getFunctionDetails().getNamespace(),
+                            instanceConfig.getFunctionDetails().getName());
+
+                    V1PodList response;
+                    try {
+                        response = coreClient.listNamespacedPod("default", null, null,
```
Comment (cckellogg) [inline pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/KubernetesRuntime.java]: This namespace should be the "jobNamespace"

Comment (jerrypeng) [reply pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/KubernetesRuntime.java]: good catch

**Comment by merlimat** [comment]
Merged in 2.3.1 at 
c579954

**Pull request merged**